scripts/terminalserver/terminalserver.jl

```diff
@@ -1,28 +1,45 @@
-
+@static if VERSION < v"1.8.0"
+    import Pkg
+end
 
 # this script basially only handles `ARGS`
 let distributed = Base.PkgId(Base.UUID("8ba89e20-285c-5b6f-9357-94700520ee1b"), "Distributed")
+    include("../error_handler.jl")
+
+    version_specific_env_path = joinpath(@__DIR__, "..", "environments", "terminalserver", "v$(VERSION.major).$(VERSION.minor)")
+    if !isdir(version_specific_env_path)
+        version_specific_env_path = joinpath(@__DIR__, "..", "environments", "terminalserver", "fallback")
+    end
+
+    @static if VERSION < v"1.8.0"
+        Pkg.activate(version_specific_env_path)
+    else
+        Base.set_active_project(joinpath(version_specific_env_path,"Project.toml"))
+    end
+
+
+    args = [popfirst!(Base.ARGS) for _ in 1:9]
+    conn_pipename, debug_pipename, telemetry_pipename, project_path = args[1:4]
+
     if haskey(Base.loaded_modules, distributed) && (Distributed = Base.loaded_modules[distributed]).nprocs() > 1
+        # TODO Move this logic over to the project loading mechanism
```

Comment:
@pfitzseb Not sure what we need to do here. Maybe do a `remotecall_eval` to activate the project everywhere, then do the `using` call, then do another `remotecall_eval` to activate the user project everywhere? I assume activating projects in the main process doesn't automatically propagate to the child workers?